### DISCUSSION TITLE
Document an officially recommended `.gitignore` for direnv users?

### DISCUSSION THREAD

**Discussion by spenserblack** [direnv/direnv#1287] [Ideas]
Just started using this tool and I love it!

I think it's pretty evident that the `.direnv` folder, which can contain [virtual environments and dependencies](https://github.com/direnv/direnv/blob/f8ab875377c288fa2179b99bc9393ba039ac8ec0/man/direnv-stdlib.1.md?plain=1#L264-L276) should be ignored. Even though it might be obvious, however, a documented recommended `.gitignore` can be helpful.

Context: I think it would be helpful for the github/gitignore repository to have an entry for direnv in their [`Global`](https://github.com/github/gitignore/tree/3674ae092f8055c29ffac2066dce1ccd7f798891/Global) folder. To move that forward, it would be helpful to be able to point to official documentation as justification.

**Comment by pitoniak32** [comment]
I have started using direnv, with sensitive and non sensitive values, and recently have been working on the idea of using a "secret manager" (https://github.com/pitoniak32/dotenv_gsm) for the sensitive values. 

My `.gitignore` looks like this:
```
.env*
!.env/.env.gsm
!.env/.env.non-secret
```

This allows for a secrets file `.env/.env.secret` to be excluded from VCS, and prevents any non explicitly defined files from being checked in. But also allows for non sensitive config to be shared between developers.

`.envrc`:
```
# DO NOT ADD VARIABLES HERE

dotenv .env/.env.non-secret

[[ -f .env/.env.secret ]] && dotenv .env/.env.secret

if [ -f .env/.env.gsm ]; then
  dotenv .env/.env.gsm

  if has dotenv_gsm; then
    eval "$(LOG_LEVEL=debug dotenv_gsm bash .env/.env.gsm)"
  fi
fi
```

And then I have a special env file `.env/.env.gsm` that gets loaded by my "plugin" for direnv `dotenv_gsm`. It pulls values from Google Secret Manager and makes them available in the environment:
```
SECRET_VAR='projects/test-project/secrets/my-super-secret'
```

**Reply by pitoniak32** [reply]
I wanted to start a discussion about if theres an easier way to do this kind of thing already in `direnv` where I want to watch an env file, but pass the output of the `dotenv` cmd to an intermediate command before it gets exported to the environment? Something like `dotenv .env/.env.gsm | dotenv_gsm` because I know that the `dotenv` built in will output `export VAR='value'`, I just want to do some processing on it before it gets exported. Because currently the `dotenv_gsm` is doing lots of duplicate work and had to copy in code from this project 😓

**Reply by pitoniak32** [reply]
right after I posted this message I realized how this should actually be done 🙌🏼
```bash
# --- Contents of: `~/.config/direnv/direnvrc` ---

# Usage: dotenv_gsm [<dotenv>]
#
# Loads a ".env.gsm" file and fetches values from Google Secret Manager
dotenv_gsm() {
  local path=${1:-}
  if [[ -z $path ]]; then
    path=$PWD/.env
  elif [[ -d $path ]]; then
    path=$path/.env
  fi
  watch_file "$path"
  if ! [[ -f $path ]]; then
    log_error ".env at $path not found"
    return 1
  fi
  if has ./dist/dotenv-gsm_darwin_amd64_v1/dotenv-gsm; then
    eval "$("$direnv" dotenv bash "$@" | LOG_LEVEL=info ./dist/dotenv-gsm_darwin_amd64_v1/dotenv-gsm -)"
  else
    log_error "ERROR: ensure you have installed 'dotenv-gsm', and its available on your path!"
    return 1
  fi
}

# --- Contents of: `.envrc` ---

dotenv_gsm .env/.env.gsm
```
